master/internal/api_experiment.go

```diff
@@ -840,7 +914,13 @@ func (a *apiServer) PatchExperiment(
 		}
 	}
 
-	return &apiv1.PatchExperimentResponse{Experiment: exp}, nil
+	// include queued / pulling / starting / running state
+	expListForm, err := a.enrichExperimentState([]*experimentv1.Experiment{exp})
```

Comment:
since this takes a pointer and mutates the experiment anyway, you could just not make it 'appear' functional and also make it take a variadic number of experiments, solving this weird slice pass/slice get thing

Reply:
on this and `getExperiment` (lowercase) I changed it to pass by reference and not use `expListForm`, seems to work